cmd/nodepool/openstack/create.go

```diff
@@ -117,8 +153,37 @@ func (o *OpenStackPlatformCreateOptions) Type() hyperv1.PlatformType {
 }
 
 func (o *OpenStackPlatformCreateOptions) NodePoolPlatform() *hyperv1.OpenStackNodePoolPlatform {
-	return &hyperv1.OpenStackNodePoolPlatform{
-		Flavor:    o.Flavor,
-		ImageName: o.ImageName,
+	nodePool := &hyperv1.OpenStackNodePoolPlatform{
+		Flavor:          o.Flavor,
+		ImageName:       o.ImageName,
+		AdditionalPorts: o.AdditionalPorts,
+	}
+
+	return nodePool
+}
+
+func validateAdditionalPorts(additionalPorts []string) error {
+	for _, additionalPortOptsRaw := range additionalPorts {
+		additionalPortOpts := PortOpts{}
+		if err := params.Map("openstack-node-additional-ports", additionalPortOptsRaw, &additionalPortOpts); err != nil {
+			return err
+		}
+		if additionalPortOpts.NetworkID == "" {
+			return fmt.Errorf("--openstack-node-additional-ports requires network-id to be set")
+		}
+	}
+	return nil
+}
+
+func strToBool(s string) bool {
+	return s == "true"
+}
+
+func getAddressPairs(addressPairs string) []hyperv1.AddressPair {
+	separetedAddressPairs := strings.Split(addressPairs, "-")
```

Comment:
this is the only separator that I could make work and thought of that wasn't that bad...